core/templates/dev/head/css/oppia.css

```diff
@@ -1195,11 +1195,34 @@ pre.oppia-pre-wrapped-text {
   border-radius: 4px;
 }
 
+.oppia-editable-section:hover pre {
+	background-color: inherit;
+}
+
+.oppia-editable-section:hover oppia-noninteractive-link,
```

Comment:
Ok @seanlip, After reading all comments seems like I will have to change the whole structure. 
I think of deleting this PR and creating a new one from start. 
If there is no problem, then can you please delete this one ?